src/domain/task/store/reducer/task.reducer.ts

```diff
@@ -68,10 +73,57 @@ const removeTask = (
   }
 }
 
+const configureTaskProgressBounds = (
+  state: DeepReadonly<TaskState>,
+  payload: DeepReadonly<TaskProgressBoundsConfiguredPayload>
+): TaskState => {
+  const { id, timestamp, progressBounds }: TaskProgressBoundsConfiguredPayload = payload
+  const foundTaskById = state.byId.get(id)
+  const { max, min }: Progress = progressBounds
+  const bounds = {
```

Comment:
There is a subtlety here. By doing this, we can update the bounds with concrete values, but we can never remove them, while your invariant accepts it. So we have an unattainable state.

It's not easy to manage, and I think we have to think a little bit before changing the code, which can be addressed in a subsequent PR.

Reply:
Yes, it's a tricky interrogation.
I see your point, but I think we could address it later because that modifies the way we use commands and we design the store.

Reply:
Of course!